garnett/context.py

```diff
@@ -1,23 +1,17 @@
 from contextlib import ContextDecorator
 import contextvars
 
-from garnett.utils import get_default_language
-
-ctx_language = contextvars.ContextVar("garnett_language")
+# Internal context var should be set via set_field_language and get via get_current_language
+_ctx_language = contextvars.ContextVar("garnett_language")
 
 
 class set_field_language(ContextDecorator):
-    def __init__(self, language, deactivate=False):
-        self.old_language = ctx_language.set(language).old_value
-        if self.old_language == contextvars.Token.MISSING:
-            self.old_language = get_default_language()
-        self.deactivate = deactivate
+    def __init__(self, language):
+        self.language = language
+        self.token = None
 
     def __enter__(self):
-        pass
+        self.token = _ctx_language.set(self.language)
 
     def __exit__(self, exc_type, exc_value, traceback):
-        # Using Context.reset like shown below causes weird errors
-        # TODO: Maybe fix this?
-        # ctx_language.reset(self.language)
-        ctx_language.set(self.old_language)
+        _ctx_language.reset(self.token)
```

Comment:
If this works - awesome, but I had difficulties getting it to work

Reply:
I have tested this and it's working nicely. Problem was you need to pass the output of set directly to reset